platform/keysystem/keyserver/config/context.go

```diff
@@ -16,16 +16,10 @@ type StaticFile struct {
 	Filepath string
 }
 
-type Grant struct {
-	API                string
-	Group              *account.Group
-	PrivilegeByAccount map[string]account.Privilege
-}
-
 type Context struct {
 	Authorities             map[string]authorities.Authority
 	Groups                  map[string]*account.Group
-	Grants                  map[string]Grant
+	Grants                  map[string]map[string]account.Privilege // indexed first by API and then by principal
```

Comment:
Agreed. I'm definitely interested in reducing the complexity of the access management system involved here, and I think simplifying this will fall out naturally from our refactoring process.